corehq/apps/data_interfaces/models.py

```diff
@@ -49,18 +61,26 @@ def organize_rules_by_case_type(cls, rules):
 
     @classmethod
     def get_boundary_date(cls, rules, now):
```

Comment:
should add a comment that says it returns None if any of the rules don't filter on server modified